### DISCUSSION TITLE
Data security - can't get my head around. Help pls

### DISCUSSION THREAD

**Discussion by jesperordrup** [pocketbase/pocketbase#5626] [Q&A]
I'm really struggling with this one. How can I expand a relation to a user table and be sure not to create a possible data slip?

Example

Collection "clients" have a relation "contact" to users.

Anyone is allowed to view clients and should be able to view contact email and name.

I'm logged in as a ordinary user with no access but to my own user record both serverside and clientside (sveltekit)

I want to expand the client table contact relation so I must use:

?expand=contact

But for that to work I have to adjust the apirules to allow all to view/list all users?

But is that really a good idea?

And if I do that, what prevents a clientside request for collection users to get all users? 

Alternative 
Since I'm serverside I could:

Login as admin
Fetch users
Login as user
Fetch clients
Map

And thereby keep users API rules intact.

But that can't be the right way?

Side question: 

Wouldn't it be great if you could have a relation to a view?

**Comment by ganigeorgiev** [comment]
It is not clear what is the desired behavior.

In order to be able to expand a relation the client request must satisfy the View API rule of the expanded rel collection.

Whether granting this access is a good idea or not - depends on your use case.

Note that the users `email` field is special and it'll be returned only if:

- the authenticated client is an admin ("superuser" with v0.23)
- the authenticated client is the owner of the auth record (aka. `@request.auth.id = id`)
- the user has explicitly allowed their email to be public (aka. `emailVisibility` is set to `true`)


If you have other sensitive fields in your users collection and don't want other users to have access to them you can move them to a separate collection that links back to the main one with a single `relation` field.


> Wouldn't it be great if you could have a relation to a view?

It is deliberately not allowed because it is difficult to generalize and may not always work (see https://github.com/pocketbase/pocketbase/issues/3000).

**Comment by jesperordrup** [comment]
> If you have other sensitive fields in your users collection and don't want other users to have access to them you can move them to a separate collection that links back to the main one with a single relation field.

Of course - that works

> It is deliberately not allowed because it is difficult to generalize and may not always work

Using views with out cascade support would be awesome and also usefull imho to have a relation that is filtered.
